runtime-sdk/modules/contracts/src/abi/oasis/memory.rs

```diff
@@ -46,6 +46,26 @@ impl Region {
         }
     }
 
+    /// Dereferences a pointer to the region.
+    pub fn deref(memory: &wasm3::Memory<'_>, arg: u32) -> Result<Self, RegionError> {
+        let arg = arg as usize;
+
+        // Make sure the pointer is within WASM memory.
+        if arg as usize >= memory.size() {
```

Comment:
This should be `arg + 8` (e.g. `arg + sizeof(HostRegion)`).

Reply:
yeah ofcourse